File: Android/MyApplication2/app/src/main/java/com/example/gold/recyclerviewstudy/Adapter.java
package com.example.gold.recyclerviewstudy;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Gold on 2017. 3. 18..
 */

public class Adapter extends RecyclerView.Adapter<Adapter.ViewHolder> {

    List<Student> items = new ArrayList<>();
    View view;


    @Override
    public Adapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        view = LayoutInflater.from(parent.getContext()).inflate(R.layout.list_item, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(Adapter.ViewHolder holder, int position) {
        Student item = items.get(position);
        holder.txt1.setText(item.getName());
        holder.txt2.setText(item.getScore());
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        TextView txt1, txt2;

        public ViewHolder(View itemView) {
            super(itemView);
            txt1 = (TextView) itemView.findViewById(R.id.txt1);
            txt2 = (TextView) itemView.findViewById(R.id.txt2);
        }
    }
}

File: Android/KotlinStudy/app/src/main/java/com/android/gold/kotlinstudy/Adapter.kt
package com.android.gold.kotlinstudy

import android.content.Context
import android.support.v4.content.ContextCompat
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

/**
 * Created by Gold on 2017. 11. 15..
 */
class Adapter(val context: Context, val items: List<NationData>) : RecyclerView.Adapter<ViewHolder>()
{

	private var onItemClick: View.OnClickListener? = null

	override fun onBindViewHolder(holder: ViewHolder, position: Int) {
		holder.img_flag.setImageResource(R.drawable.ic_launcher_foreground)
		holder.txt_name.setText("")
	}

	override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): ViewHolder? {
		val mainView = LayoutInflater.from(context).inflate(R.layout.list_item, parent, false)
		return ViewHolder(mainView)
	}

	override fun getItemCount(): Int {
		TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
	}
}

class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
	val img_flag = view.findViewById<View>(R.id.img_flag) as ImageView
	val txt_name = view.findViewById<View>(R.id.text_name) as TextView
	val txt_capital = view.findViewById<View>(R.id.capital) as TextView
}

data class NationData (
		var resId:Int,
		var name: String,
		var capital: String
)